packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -921,21 +948,25 @@ class _BottomNavigationBarState extends State<BottomNavigationBar> with TickerPr
     switch (_effectiveType) {
       case BottomNavigationBarType.fixed:
         colorTween = ColorTween(
-          begin: widget.unselectedItemColor
+          begin: effectiveUnselectedLabelStyle.color
+            ?? widget.unselectedItemColor
             ?? bottomTheme.unselectedItemColor
             ?? themeData.unselectedWidgetColor,
-          end: widget.selectedItemColor
+          end: effectiveSelectedLabelStyle.color
+            ?? widget.selectedItemColor
             ?? bottomTheme.selectedItemColor
             ?? widget.fixedColor
             ?? themeColor,
```

Comment:
This `ColorTween` is a bit odd in that it is used for both the icon and label colors, even though they can be different colors. I know this was already in place before your fix, but this just seems to conflate the two even more. Perhaps we should break these out into a tween for the icon and a tween for the text.

Reply:
@darrenaustin So, if i make the previous changes without changing this color tween, all tests are successful, as soon I add a separate tween for the icon colors some golden tests fail.